packages/messenger-widget/src/components/RightHeader/RightHeader.tsx

```diff
@@ -53,58 +69,135 @@ export function RightHeader(props: HideFunctionProps) {
         }
     };
 
+    // Method to open 3 dot icon menu to in mobile screen
+    const openMenu = () => {
+        const menu = document.querySelector('.dropdown-content');
+        if (menu && !menu.classList.contains('menu-details-dropdown-content')) {
+            menu.classList.add('menu-details-dropdown-content');
+        } else {
+            closeContactMenu();
+        }
+    };
+
     // loads the profile pic on page render
     useEffect(() => {
         fetchAndSetProfilePic();
     }, []);
 
     return (
-        <div
-            className={(props.showContacts
-                ? 'justify-content-end'
-                : 'justify-content-between'
-            ).concat(
-                ' col-12 d-flex align-items-center pr-0 profile-name-container'.concat(
-                    ' ',
-                    state.uiView.selectedRightView === RightViewSelected.Profile
-                        ? ' background-chat'
-                        : ' background-container',
-                ),
-            )}
-        >
-            {!props.showContacts && (
+        <>
+            {/* Header for mobile screen */}
+            {screenWidth <= MOBILE_SCREEN_WIDTH ? (
```

Comment:
Can we split this part into two components (MobileView, NormalView, or something like that)? This would keep the concerns separate and increase maintainability